primitives/compile.futil

```diff
@@ -13,6 +13,11 @@ comb primitive std_wire<"share"=1>[WIDTH](@data in: WIDTH) -> (out: WIDTH) {
     assign out = in;
 }
 
+/// Wire specifically for instrumentation
+primitive std_inst_wire[WIDTH](in: WIDTH) -> (out: WIDTH) {
+    assign out = in;
+}
+
```

Comment:
This file should only contain primitives that are used for compilation and in general should not be edited unless you're changing the compilation process.

Can you talk a bit more about why using `std_wire` and marking every _instance_ of `std_wire` generated by the instrument pass cannot be marked with `@protected` to get the same effect?

Reply:
I changed the primitive around a bit (now named `std_probe`) so that it's now more distinct from `std_wire`! When I wrote the PR this new primitive seemed a lot like `std_wire` since that's where we started from, but I think we actually want a primitive specifically for profiling.

I put the primitive in `compile.futil` since in theory cells of the primitive should only be introduced in the `instrumentation` pass. Would that be ok?

Reply:
Can you discuss a bit how this primitive is used and why a primitive with `@protected` + an attribute for instrumentation how sufficient? I think in the hierarchy of things added to the compiler, it is better to add an attribute over a new cell.

Finally, if we have to add a cell, let's name this primitive `std_sink` (since it seems to act like an signal sink). `compile.futil` still seems sketch; the requirement for this pass is primitives that are required for compilation, not primitives that are only added by compilation (for example, anyone can use `std_wire`). `compile.futil` gets bundles into the source of the compiler so the bar should be high.

Reply:
Nvm about the new primitive, we figured out how to get `@protected` to preserve a `std_wire` during compilation, so we should be good now!